backend/PythonClient/server/simulation_server.py

```diff
@@ -33,90 +36,53 @@
 
 @app.route('/list-reports', methods=['GET'])
 def list_reports():
-    # Reports file
-    reports_path = os.path.join(os.path.expanduser("~"), "Documents", "AirSim", "report")
-    if not os.path.exists(reports_path) or not os.path.isdir(reports_path):
-        return 'Reports directory not found', 404
-    def check_monitor_pass_fail(directory, monitor):
-        monitor_pass = True
-        monitor_path = os.path.join(directory, monitor)
-        if os.path.exists(monitor_path) and os.path.isdir(monitor_path):
-            for file_name in os.listdir(monitor_path):
-                if file_name.endswith('.txt'):
-                    log_path = os.path.join(monitor_path, file_name)
-                    with open(log_path, 'r') as file:
-                        for line in file:
-                            if 'FAIL' in line:
-                                monitor_pass = False
-                                break
-                    if not monitor_pass:
-                        break
-        return monitor_pass
-    def count_drone_files(directory):
-        flytopoints_path = os.path.join(directory, 'FlyToPoints')
-        if os.path.exists(flytopoints_path) and os.path.isdir(flytopoints_path):
-            collision_monitor_path = os.path.join(flytopoints_path, 'CollisionMonitor')
-            if os.path.exists(collision_monitor_path) and os.path.isdir(collision_monitor_path):
-                return len([f for f in os.listdir(collision_monitor_path) if f.endswith('.txt')])
-        return 0
+    # Fetch the list of reports from GCS bucket
+    bucket = storage_client.bucket(bucket_name)
+    blobs = bucket.list_blobs(prefix='reports/')
```

Comment:
The `list_reports` function needs to properly handle GCS's flat namespace by using the `prefix` and `delimiter` parameters in `list_blobs`. The current code incorrectly assumes a local filesystem structure.